packages/cozy-procedures/src/components/Overview.jsx

```diff
@@ -12,32 +16,115 @@ import {
 
+import Topbar from './Topbar'
+
 class Overview extends React.Component {
+  realtime = null
+
+  componentWillUnmount() {
+    if (this.realtime) {
+      this.realtime.unsubscribeAll()
+    }
+  }
+
   navigateTo = view => {
     const { location, router } = this.props
     const rootPath = location.pathname
     const separator = rootPath.endsWith('/') ? '' : '/'
     router.push(`${rootPath}${separator}${view}`)
   }
 
+  ensureRealtime = () => {
+    if (!this.realtime) {
+      this.realtime = new CozyRealtime({ client: this.props.client })
+    }
+  }
+
+  handleZipChanged = jsonFile => job => {
+    const { client, t } = this.props
+    if (job.state === 'done') {
+      Alerter.success(t('overview.zip_ready'))
+      client.collection('io.cozy.files').destroy(jsonFile)
+    }
+  }
+
+  getDestinationId = async template => {
+    const { client } = this.props
+    return await client
+      .collection('io.cozy.files')
+      .ensureDirectoryExists(template.pathToSave)
+  }
+
+  submitProcedure = async () => {
+    const { client } = this.props
+    const template = creditApplicationTemplate
+    try {
+      const destinationId = await this.getDestinationId(template)
+      const datetime = new Date().toISOString()
+      const baseFilename = `${template.type}-${datetime}`
+      const jsonFilename = `${baseFilename}.json`
+      const administrativeProcedure = AdministrativeProcedure.create(
+        this.props.data,
+        creditApplicationTemplate
+      )
+      const response = await client.create(
+        'io.cozy.procedures.administratives',
+        administrativeProcedure
+      )
+      const jsonData = AdministrativeProcedure.createJson(response.data)
+
+      const file = new File([jsonData], jsonFilename)
+      const resp = await client
+        .collection('io.cozy.files')
+        .createFile(file, { dirId: destinationId })
+
+      const jsonFile = resp.data
+      const params = {
+        files: {
+          [jsonFilename]: jsonFile.id
+          // TODO: use real files from redux state
+          // 'documents/some.jpg': 'fd3b80df50114b26a723700b7006deda'
+        },
+        dir_id: destinationId,
+        filename: `${baseFilename}.zip`
+      }
+      this.ensureRealtime()
+      const zipResp = await client
+        .collection('io.cozy.jobs')
+        .create('zip', params)
+
+      await this.realtime.subscribe(
+        'updated',
+        'io.cozy.jobs',
+        zipResp.data.id,
+        this.handleZipChanged(jsonFile)
+      )
+    } catch (e) {
+      console.error(e)
+    }
+  }
+
   render() {
     const {
       personalDataFieldsCompleted,
       personalDataFieldsTotal,
-      amount,
-      duration,
+      data,
       t
     } = this.props
+    const { amount, duration } = data.procedureData
     return (
       <div>
         <Topbar title={creditApplicationTemplate.name} />
         <Title className="u-mb-2">{t('overview.subtitle')}</Title>
+        <Alerter />
```

Comment:
That might be a problem if there's already an Alerter instance in the app's tree :/

Reply:
How would you "fix" this? 
We could either :
- say it's the app's responsibility to display the `Alerter` component
- add a prop like `withAlerter` to know if we want to display it (and override this prop if we don't want to)

Reply:
I think 1 is ok, especially since we might move away from alert's in the near future.